forms-flow-api-utils/src/formsflow_api/config.py

```diff
@@ -0,0 +1,206 @@
+"""All of the configuration for the api is captured here.
+
+All items are loaded, or have Constants defined here that
+are loaded into the Flask configuration.
+All modules and lookups get their configuration from the
+Flask config, rather than reading environment variables directly
+or by accessing this configuration directly.
+"""
+
+import os
+
+from dotenv import find_dotenv, load_dotenv
+
+# this will load all the envars from a .env file located in the project root (api)
+load_dotenv(find_dotenv())
+
+CONFIGURATION = {
+    "development": "formsflow_api.config.DevConfig",
+    "testing": "formsflow_api.config.TestConfig",
+    "production": "formsflow_api.config.ProdConfig",
+    "default": "formsflow_api.config.ProdConfig",
+}
+
+
+def get_named_config(config_name: str = "production"):
+    """Return the configuration object based on the name.
+
+    :raise: KeyError: if an unknown configuration is requested
+    """
+    if config_name in ["production", "staging", "default"]:
+        config = ProdConfig()
+    elif config_name == "testing":
+        config = TestConfig()
+    elif config_name == "development":
+        config = DevConfig()
+    else:
+        raise KeyError(f"Unknown configuration '{config_name}'")
+    return config
+
+
+class _Config:  # pylint: disable=too-few-public-methods
+    """Base class configuration.
+
+    that should set reasonable defaults for all the other configurations.
+    """
+
+    PROJECT_ROOT = os.path.abspath(os.path.dirname(__file__))
+
+    SECRET_KEY = "secret value"
+
+    SQLALCHEMY_TRACK_MODIFICATIONS = False
+
+    ALEMBIC_INI = "migrations/alembic.ini"
+
+    # POSTGRESQL
+    SQLALCHEMY_DATABASE_URI = os.getenv("DATABASE_URL", "")
+
+    TESTING = False
+    DEBUG = False
+
+    # JWT_OIDC Settings
+    JWT_OIDC_WELL_KNOWN_CONFIG = os.getenv("JWT_OIDC_WELL_KNOWN_CONFIG")
+    JWT_OIDC_ALGORITHMS = os.getenv("JWT_OIDC_ALGORITHMS", "RS256")
+    JWT_OIDC_JWKS_URI = os.getenv("JWT_OIDC_JWKS_URI")
+    JWT_OIDC_ISSUER = os.getenv("JWT_OIDC_ISSUER")
+    JWT_OIDC_AUDIENCE = os.getenv("JWT_OIDC_AUDIENCE")
+    JWT_OIDC_CACHING_ENABLED = os.getenv("JWT_OIDC_CACHING_ENABLED")
+    JWT_OIDC_JWKS_CACHE_TIMEOUT = 300
+
+    # Keycloak Service for BPM Camunda
```

Comment:
Please remove unwanted config params to make the deployment templates simple.
So far, we would need only JWT + Form io config I guess ?